runtime/interp.c

```diff
@@ -64,6 +64,8 @@ sp is a local copy of the global variable Caml_state->extern_sp. */
 #  else
 #    define Next goto *(void *)(jumptbl_base + *pc++)
 #  endif
+#  undef fallthrough
+#  define fallthrough ((void) 0)
```

Comment:
Nitpick: I find it arcane to change the semantics of `fallthrough` in just this field, I would rather introduce a `Continue` macro (or `Fallthrough` or something else than `fallthrough`) that is defined as `(void) 0` in the threaded case and `fallthrough` otherwise.